src/FlatSharp.Compiler/MetadataHelpers.cs

```diff
@@ -29,40 +27,45 @@ public static class MetadataHelpers
             string key,
             TryParse<T> tryParse,
             T defaultValueIfPresent,
-            T defaultIfNotPresent)
+            T defaultValueIfNotPresent)
         {
-            if (metadata.TryGetValue(key, out string value))
+            if (!metadata.TryGetValue(key, out var value))
             {
-                if (string.IsNullOrEmpty(value))
-                {
-                    return defaultValueIfPresent;
-                }
-                else if (tryParse(value, out T parsed))
-                {
-                    return parsed;
-                }
-                else
-                {
-                    ErrorContext.Current?.RegisterError($"Unable to parse attribute '{key}' with value '{value}' as as type '{typeof(T).Name}'. ");
-                    return defaultValueIfPresent;
-                }
+                return defaultValueIfNotPresent;
             }
-            else
+
+            if (string.IsNullOrEmpty(value))
             {
-                return defaultIfNotPresent;
+                return defaultValueIfPresent;
             }
+
+            if (tryParse(value, out var parsed))
+            {
+                return parsed;
+            }
+
+            ErrorContext.Current?.RegisterError(
+                $"Unable to parse attribute '{key}' with value '{value}' as as type '{typeof(T).Name}'. ");
+
+            return defaultValueIfPresent;
         }
 
         public static bool ParseBooleanMetadata(this IDictionary<string, string> metadata, string key)
         {
-            return ParseMetadata<bool>(
+            return ParseMetadata(
                 metadata,
                 key,
                 bool.TryParse,
                 true,
                 false);
         }
 
+        public static bool TryParseIntegerMetadata(this IDictionary<string, string> metadata, string key, out int value)
+        {
+            value = ParseMetadata(metadata, key, int.TryParse, DefaultValueIfPresent, DefaultValueIfNotPresent);
+            return value >= 0;
```

Comment:
Given  that it must be non-negative to be valid, it seems like this should be a UInt instead of int, right?